docs/design.md

```diff
@@ -1,175 +1,241 @@
 # Design
 
-## Models
+## Common Functionality across all Kinds
 
-A Model object represents a logical ML model. The object describes a location
-where ML model files can be stored for a specific model. 
+### Container Images
 
-Optionally, a loader can be specified to load ML models weights and biases from specific sources.
-For example, a commonly used model loader would be the HuggingFace model loader
-where by providing the model name would load the model to the Model object
-bucket location. Model loaders are defined as container images and can either
-point to prebuilt container images or a Git repository with Dockerfile.
+All Substratus kinds correspond to actions that are taken by containers. These containers can be built
+by Substratus (from git, or from an upload) or referenced from external registries.
 
-A model
+```yaml
+spec:
+  image:
+    # Optional git source.
+    git:
+      url: https://github.com/substratusai/hf-model-loader
+    
+    # Optional upload source.
+    upload: {}
 
-The Model object can optionally describe the training parameters including a reference to the Dataset that should be used for training. In this case, the Model object represents the eventual output of a training/finetuning operation.
+    # Optionally specify image name.
```

Comment:
Should we call out that you can only set 1 of the optional parameters and are not allowed to set both git and name?